stl/inc/algorithm

```diff
@@ -274,6 +274,23 @@ namespace ranges {
         }
     }
 
+    template <forward_range _Rng>
+    _NODISCARD constexpr auto _Get_final_iterator_unwrapped(_Rng& _Range, const _Unwrapped_t<iterator_t<_Rng>>& _Mid) {
+        // find the (unwrapped) iterator in _Range which equals _Uend(_Range) [possibly O(N)]
+        if constexpr (common_range<_Rng>) {
+            return _Uend(_Range);
+        } else if constexpr (sized_range<_Rng>) {
+            const auto _Dist = _RANGES distance(_Range);
+            if constexpr (sized_sentinel_for<iterator_t<_Rng>, iterator_t<_Rng>>) {

```

Comment:
So shouldn't this test `_Unwrapped_t<iterator_t<_Rng>>`?

I believe this collapses down to `!std::disable_sized_sentinel_for<_Unwrapped_t<iterator_t<_Rng>>, _Unwrapped_t<iterator_t<_Rng>>>`

I am unsure whether it is clearer to use the concept though

Reply:
The design requires that unwrapped iterator types model all of the concepts that the corresponding wrapped iterator type models, so `sized_sentinel_for<iterator_t<_Rng>, iterator_t<_Rng>>` implies `sized_sentinel_for<_Unwrapped_t<iterator_t<_Rng>>, _Unwrapped_t<iterator_t<_Rng>>>`. That said, I do think it would be clearer to use the unwrapped type here.

`sized_sentinel_for` also requires `-` to be valid, and it refines `sentinel_for` which requires `==` and `!=` to be valid, none of which are required by `input_or_output_iterator`, so there's more there than simply `!disable_sized_sentinel_for`.